### DISCUSSION TITLE
Finer granular progress update in postprocessors

### DISCUSSION THREAD

**Discussion by FlorianBorn71** [assimp/assimp#4413] [Ideas]
Hi!
We figure there is some rudimentary support for a progress update callback but so far it only works with the granularity of full processing steps. However, in our use cases we have models where for instance tangents calculation may take ~30min. In these cases a finer granularity for the update callback would be critical.

Are there any plans to improve this?

Or, if we implemented this, would you be open for merging this into the codebase?

Cheers!
Florian Born

**Comment by kimkulling** [comment]
30 minutes is quite log. Do you have a model which can be used to reproduce this?

**Reply by FlorianBorn71** [reply]
This is customer data so I'm afraid I can't provide you with those specific models. But I can see if I can build a synthetic model to reproduce this. 

Things to note:
- our models are pretty big, sometimes >100M triangles, so the times are not unrealistic
- even if the processing step is just 2min, a finer grained progress update would be desirable. I have looked into the code and I think we could come up with a good design. My main question would thus be whether you would be open to have that feature merged eventually, of course assuming you are happy with our proposal ;-)

Cheers!
Florian
